### PR TITLE
DOC add a new section for end-to-end pipeline user guide

### PR DESCRIPTION
**Pull request by glemaitre**
This PR add a new section in the documentation to put forward the `tabular_learner` first. In the future, this will be the home of the recipe as well.

Encoders, joiners, and cleaners will come in subsequent sections as detailed discussions.

### CODE REVIEW

File: doc/end_to_end_pipeline.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+============================
+
+.. currentmodule:: skrub
+
+.. _table_vectorizer:
+
+Turning a dataframe into a numerical feature matrix
+---------------------------------------------------
+
+A dataframe can comprise columns of all kind of types. A good numerical
```
Comment (TheooJ) [inline doc/end_to_end_pipeline.rst:14]:
```suggestion
A dataframe can contain columns of all kind of types. A good numerical
```

File: doc/end_to_end_pipeline.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+representation of these columns help analytics and statistical learning.
+
+The :class:`TableVectorizer` gives a turn-key solution by applying
+different data-specific encoder to the different columns. It makes reasonable
```
Comment (TheooJ) [inline doc/end_to_end_pipeline.rst:18]:
```suggestion
different data-specific encoders to the different columns. It makes reasonable
```

File: doc/end_to_end_pipeline.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+heuristic choices that are not necessarily optimal since it is not aware of the learner
+used for the machine learning task). However, it already provides a typically very good
+baseline.
+
+The function :func:`tabular_learner` goes the extra mile by creating a machine-learning
+model that works well on tabular data. This model combines a :class:`TableVectorizer`
+with a provided scikit-learn estimator. Depending whether or not the final estimator
+natively support missing values, a missing value imputer step is added before the
```
Comment (TheooJ) [inline doc/end_to_end_pipeline.rst:26]:
```suggestion
natively supports missing values, a missing value imputer step is added before the
```

**Comment by glemaitre** [comment]
I takes those changes but I'll rework this section anyway. Here this is just a copy paste to have the right section in the table of content.

**Review by jeromedockes — APPROVED** [review]
thanks! as we said we will iterate on the section names and the individual sections in subsequent prs

**Pull request merged**